serverless/virus-scanner/src/clamscan.service.ts

```diff
@@ -0,0 +1,67 @@
+import NodeClam from 'clamscan'
+import internal, { Stream } from 'stream'
+
+import { getLambdaLogger } from './logger'
+import { ScanFileStreamResult } from './types'
+
+export async function scanFileStream(
+  s3Stream: internal.Readable,
+): Promise<ScanFileStreamResult> {
+  const logger = getLambdaLogger('scanFileStream')
+
+  logger.info('Scanning file stream')
+
+  const scanner = await new NodeClam().init({
+    clamdscan: {
+      socket: '/tmp/clamd.ctl',
+    },
+  })
+
+  // create a writable stream to memory
+  const outputStream = new Stream.Writable()
```

Comment:
I'm wondering if this is necessary since whenever we go from streams to fixed buffers in memory it becomes a resource exhaustion/DoS risk; can it just do an S3 API CopyObject so you don't have to do this? The IAM role should be strictly scoped to the file though.

Reply:
thanks for this! as discussed separately, we keep to the current setup to guarantee that the clean file being saved is the file being scanned

resource exhaustion is less of a risk as file size limit is enforced

Reply:
Actually we are going to try enabling bucket versioning + use versionId as unique version key. will circle back on this

Reply:
copy and versioning implemented in [7dea90f](https://github.com/opengovsg/FormSG/pull/6642/commits/7dea90f284efeac09a502e14f65ff1ed4dc17532), so no more use of stream -> buffer